traitsui/wx/ui_panel.py

```diff
@@ -1033,6 +1069,10 @@ def add_items(self, content, panel, sizer):
             # Save the reference to the label control (if any) in the editor:
             editor.label_control = label
 
+        if (len(self._label_enabled_whens) + len(self._label_visible_whens)) > 0:
+            for object in self.ui.context.values():
+                object.on_trait_change(lambda: self._label_when(), dispatch="ui")
```

Comment:
ideally we would call `self._label_when()` after this for loop, just like we do in the qt version.  The problem is, if we do this, if you run the following example:
```
from traits.api import HasTraits, Int, Str
from traitsui.api import View, Item, Label, HGroup

class A(HasTraits):
    x = Int(1)

    view = View(
        Label("Visible", visible_when="x != 0"),
        Label("Enabled", enabled_when="x != 0"),
        Item('x'),
        resizable=True
    )

if __name__ == '__main__':
    A().configure_traits()
```

you will _never_ be able to see a "Visible" label, even if you change `x` to not be 0.  (enabled works perfectly fine).  I think the reason is that, if we set the label to invisible right off the bat, it doesn't get included in the layout or something like this. Then everytime in the future, showing/hiding it does nothing. 

If we don't have `self._label_when()` (as is currently the case), then by default both the "Visible" and "Enabled" labels are shown at the start, (even though we shouldn't see "Visible" and "Enabled" should be greyed out).  From then on whenever anything is changed on the UI they will get synced up with their `enabled_when`/`visible_when` expressions and work as desired.

This is not ideal, but better than just not working at all.  I could not find a way to have the visible label be invisible to start, but then show up later if the `visible_when` condition ever got switched to True.

Reply:
let's convert this into an issue

Reply:
I've opened #1593